deployment/helm/balloons/templates/daemonset.yaml

```diff
@@ -28,9 +28,16 @@ spec:
     {{- with .Values.nodeSelector }}
         {{- toYaml . | nindent 8 }}
     {{- end }}
-      {{- if .Values.nri.patchRuntimeConfig }}
+      {{- if .Values.nri.runtime.patchConfig }}
       initContainers:
       - name: patch-runtime
+        {{- if (not (or (eq .Values.nri.runtime.config nil) (eq .Values.nri.runtime.config.pluginRegistrationTimeout ""))) }}
+        args:
```

Comment:
Same here, we should not probably allow setting 0 value. And also, do we really need validation for the timeouts here? Because we have the proper validation later in the chain when those args are passed as flags into the go code. But if you want to fail early, then perhaps we should keep it here.

Reply:
We do need this. Otherwise you cannot patch the configuration just to enable NRI without giving both timeouts as well, which we don't want. Also, we don't want to attempt working around that by giving default settings to the timeouts which we believe to be the right defaults just because at the time of this PR to config-manager they happened to be the defaults in NRI...

Reply:
Maybe we could use an explicit `"default"` to indicate we want to use a default value (IOW leave it untouched in the configuration), but then we can't use integers for the timeout so we can't enforce a simple range check using schema minimum and maximum. We'd need to use a `pattern` with a regexp like `^((default)|(2|3|4|5|6|7|8|9|(10)|(11)|(12)|(13)|(14)|(15))$`... for requestTimeout and a corresponding one for registrationTimeout.

Reply:
> Otherwise you cannot patch the configuration just to enable NRI without giving both timeouts as well, which we don't want.

I think we can, if you had something like this:
```yaml
{{- if .Values.nri.runtime.patchConfigg }}
initContainers:
- name: patch-runtime
  image: {{ .Values.initContainerImage.name }}:{{ .Values.initContainerImage.tag | default .Chart.AppVersion }}
  imagePullPolicy: {{ .Values.initContainerImage.pullPolicy }}
  volumeMounts:
  - name: containerd-config
    mountPath: /etc/containerd
  - name: crio-config
    mountPath: /etc/crio/crio.conf.d
  - name: dbus-socket
    mountPath: /var/run/dbus/system_bus_socket
  securityContext:
    privileged: true
  args:
  {{- if .Values.nri.runtime.config.pluginRegistrationTimeout }}
    - -nri-plugin-registration-timeout
    - {{ .Values.nri.runtime.config.pluginRegistrationTimeout }}
  {{- end }}
  {{- if .Values.nri.runtime.config.pluginRequestTimeout }}
    - -nri-plugin-request-timeout
    - {{ .Values.nri.runtime.config.pluginRequestTimeout }}
  {{- end }}
{{- end }}
```

Reply:
@fmuyassarov I ended up using the values JSON schema for limiting the timeouts and for ensuring that both (or neither) are given. The registration timeout is now limited to `5-30s` and the request timeout is `2-30s`.

That leaves us with the slightly user-unfriendly runtime check of `requestTimeout < registrationTimeout`, but I don't think the JSON schema would be expressive enough to allow expressing such a constraint, so this is the best I can do.

Reply:
> > Otherwise you cannot patch the configuration just to enable NRI without giving both timeouts as well, which we don't want.
> 
> I think we can, if you had something like this:
> 
> ```yaml
> {{- if .Values.nri.runtime.patchConfigg }}
> initContainers:
> - name: patch-runtime
>   image: {{ .Values.initContainerImage.name }}:{{ .Values.initContainerImage.tag | default .Chart.AppVersion }}
>   imagePullPolicy: {{ .Values.initContainerImage.pullPolicy }}
>   volumeMounts:
>   - name: containerd-config
>     mountPath: /etc/containerd
>   - name: crio-config
>     mountPath: /etc/crio/crio.conf.d
>   - name: dbus-socket
>     mountPath: /var/run/dbus/system_bus_socket
>   securityContext:
>     privileged: true
>   args:
>   {{- if .Values.nri.runtime.config.pluginRegistrationTimeout }}
>     - -nri-plugin-registration-timeout
>     - {{ .Values.nri.runtime.config.pluginRegistrationTimeout }}
>   {{- end }}
>   {{- if .Values.nri.runtime.config.pluginRequestTimeout }}
>     - -nri-plugin-request-timeout
>     - {{ .Values.nri.runtime.config.pluginRequestTimeout }}
>   {{- end }}
> {{- end }}
> ```

That would allow you to give just one of the timeouts which is something I don't want to allow.